### DISCUSSION TITLE
Constantly restarting the TS Server to have the types

### DISCUSSION THREAD

**Discussion by thelinuxlich** [ecyrbe/zodios#249] [Q&A]
Do you have the same issue where you're working and sometimes VSCode loses the Zodios types, turning everything into `any` and the only solution seems to be restarting the TS Server?

**Comment by ecyrbe** [comment]
No, but this happened to me in my ts-calc project.

This is likely due to memory+compute issue in ts. 

I hope ts 5.0 in january fixes this kind of issues.
I'll also try to do some  ts inspect/profiling when i have time to try to reduce ts type complexity in zodios

**Reply by thelinuxlich** [reply]
I tried everything, currently running VSCode Insiders + Typescript 5.0nightly seems to alleviate a bit, but still annoying. Does anyone uses other IDEs with TSServer and can report if they have the same behavior?

**Reply by thelinuxlich** [reply]
Btw, how can you profile type complexity?

**Reply by ecyrbe** [reply]
it's definitly not an easy task, tooling is really bad, but exists. You can check this: https://github.com/microsoft/TypeScript/wiki/Performance#performance-tracing

**Reply by thelinuxlich** [reply]
Wish Microsoft extracted their issue bot that does performance comparisons into a nice package :(

**Reply by thelinuxlich** [reply]
I was having the same issue with Kysely(a typed query builder) where I defined my schema with Zod and extracted the types with `z.extendShape<z.infer<typeof schema>, CreatedAtType>`. I read a lot of articles about Typescript instantiantion depth limits and found a little helper to expand the type in this case:

```ts
type LMerge<T1, T2> = {
  [k in keyof T1]: k extends keyof T2 ? T2[k] : T1[k]
} extends infer O ? { [K in keyof O]: O[K] } : never;
```
z.extendShape wasn't flattening the type so replacing with `LMerge<z.infer<typeof schema>, CreatedAtType>` made some errors like "Type instantiation is excessively deep and possibly infinite" disappear and the typechecking performance overall increased. I wonder if something similar might be applied to Zodios, especially with the type generated from ZodiosContext `ctx.app(api)`
